types/xdg_shell/wlr_xdg_surface.c

```diff
@@ -310,22 +309,14 @@ static void xdg_surface_handle_surface_commit(struct wl_listener *listener,
 	}
 }
 
-static void surface_commit_state(struct wlr_xdg_surface *surface,
-		struct wlr_xdg_surface_state *state) {
-	surface->configure_serial = state->configure_serial;
-	if (!wlr_box_empty(&state->geometry)) {
-		surface->geometry = state->geometry;
-	}
-}
-
 void handle_xdg_surface_commit(struct wlr_surface *wlr_surface) {
 	struct wlr_xdg_surface *surface =
 		wlr_xdg_surface_from_wlr_surface(wlr_surface);
 	if (surface == NULL) {
 		return;
 	}
 
-	surface_commit_state(surface, &surface->pending);
+	surface->current = surface->pending;
```

Comment:
As `has_geometry` isn't checked here, that field could be removed.

Reply:
I think I've removed all occurrences?

Reply:
Sorry, somehow didn't notice that.